Nodejs/Product/Nodejs/Workspace/PackageJsonTestContainer.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Microsoft.VisualStudio.TestWindow.Extensibility;
+using Microsoft.VisualStudio.TestWindow.Extensibility.Model;
+
+namespace Microsoft.NodejsTools.Workspace
+{
+    public sealed class PackageJsonTestContainer : ITestContainer
+    {
+        private int version;
+
+        public PackageJsonTestContainer(ITestContainerDiscoverer discoverer, string source, string testRoot)
+        {
+            this.Discoverer = discoverer;
+            this.Source = source;
+            this.TestRoot = testRoot;
+        }
+
+        private PackageJsonTestContainer(PackageJsonTestContainer other) :
+            this(other.Discoverer, other.Source, other.TestRoot)
+        {
+            this.version = other.version;
+        }
+
+        public ITestContainerDiscoverer Discoverer { get; }
+
+        public string Source { get; }
+
+        public string TestRoot { get; }
+
+        public IEnumerable<Guid> DebugEngines { get { yield break; } }
+
+        public FrameworkVersion TargetFramework { get; } = FrameworkVersion.None;
+
+        public Architecture TargetPlatform { get; } = Architecture.X86;
+
+        public bool IsAppContainerTestContainer { get; } = false;
+
+        public int CompareTo(ITestContainer other)
+        {
+            if (other == null || !(other is PackageJsonTestContainer testContainer))
+            {
+                return -1;
```

Comment:
>-1 [](start = 23, length = 2)

This doesn't actually happen, does it?  If it does, it seems unlikely that the comparison produces a consistent result when the operands are reversed.